java/iot3/mobility/src/main/java/com/orange/iot3mobility/Utils.java

```diff
@@ -0,0 +1,609 @@
+            return true;
+        } else if (d3 == 0 && onSegment(p1, q1, p2)) {
+            return true;
+        } else if (d4 == 0 && onSegment(p1, q1, q2)) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public static double direction(LatLng pi, LatLng pj, LatLng pk) {
+        return (pk.getLongitude() - pi.getLongitude()) * (pj.getLatitude() - pi.getLatitude()) - (pj.getLongitude() - pi.getLongitude()) * (pk.getLatitude() - pi.getLatitude());
+    }
+
+    public static boolean onSegment(LatLng pi, LatLng pj, LatLng pk) {
+        return Math.min(pi.getLongitude(), pj.getLongitude()) <= pk.getLongitude() && pk.getLongitude() <= Math.max(pi.getLongitude(), pj.getLongitude()) &&
+                Math.min(pi.getLatitude(), pj.getLatitude()) <= pk.getLatitude() && pk.getLatitude() <= Math.max(pi.getLatitude(), pj.getLatitude());
+    }
+
+    // Returns true if the segments intersect, otherwise false. In addition, if the lines
+    // intersect the intersection point may be stored in the doubles ilat and ilng.
+    public static boolean doSegmentsIntercept(double p0lat, double p0lng, double p1lat, double p1lng,
+                               double p2lat, double p2lng, double p3lat, double p3lng) {
+        double ilat, ilng;
+
+        double s1lat, s1lng, s2lat, s2lng;
+        s1lat = p1lat - p0lat;     s1lng = p1lng - p0lng;
+        s2lat = p3lat - p2lat;     s2lng = p3lng - p2lng;
+
+        double s, t;
+        s = (-s1lng * (p0lat - p2lat) + s1lat * (p0lng - p2lng)) / (-s2lat * s1lng + s1lat * s2lng);
+        t = ( s2lat * (p0lng - p2lng) - s2lng * (p0lat - p2lat)) / (-s2lat * s1lng + s1lat * s2lng);
+
+        if (s >= 0 && s <= 1 && t >= 0 && t <= 1)
+        {
+            // Collision detected
+            ilat = p0lat + (t * s1lat);
+            ilng = p0lng + (t * s1lng);
+
+            return true;
+        }
+
+        return false; // No collision
+    }
+
+    // Returns true if the segments intersect, otherwise false. In addition, if the lines
+    // intersect the intersection point may be stored in the doubles ilat and ilng.
+    public static LatLng getSegmentsIntercept(double p0lat, double p0lng, double p1lat, double p1lng,
+                                              double p2lat, double p2lng, double p3lat, double p3lng) {
+        double ilat, ilng;
+
+        double s1lat, s1lng, s2lat, s2lng;
+        s1lat = p1lat - p0lat;     s1lng = p1lng - p0lng;
+        s2lat = p3lat - p2lat;     s2lng = p3lng - p2lng;
+
+        double s, t;
+        s = (-s1lng * (p0lat - p2lat) + s1lat * (p0lng - p2lng)) / (-s2lat * s1lng + s1lat * s2lng);
+        t = ( s2lat * (p0lng - p2lng) - s2lng * (p0lat - p2lat)) / (-s2lat * s1lng + s1lat * s2lng);
+
+        if (s >= 0 && s <= 1 && t >= 0 && t <= 1)
+        {
+            // Collision detected
+            ilat = p0lat + (t * s1lat);
+            ilng = p0lng + (t * s1lng);
+
```

Comment:
[qodana]
Values ilng and ilat are assigned but not used
Probable bugs Unused assignment

Reply:
Method is not used and will be deleted.